src/Evolution/DgSubcell/SubcellOptions.hpp

```diff
@@ -120,20 +127,46 @@ class SubcellOptions {
         "phenomenon."};
   };
 
+  /// \brief A list of block names on which to never do subcell.
+  ///
+  /// Can overlap with blocks listed in `OnlyDgBlockGroups`.
+  ///
+  /// Set to `None` to allow subcell in all blocks.
+  struct OnlyDgBlocksAndGroups {
+    using type =
+        Options::Auto<std::vector<std::string>, Options::AutoLabel::None>;
+    static constexpr Options::String help = {
+        "A list of block and group names on which to never do subcell.\n"
+        "Set to 'None' to not restrict where FD can be used."};
+  };
+
   using options =
       tmpl::list<InitialDataRdmpDelta0, InitialDataRdmpEpsilon, RdmpDelta0,
                  RdmpEpsilon, InitialDataPerssonExponent, PerssonExponent,
-                 AlwaysUseSubcells, SubcellToDgReconstructionMethod, UseHalo>;
+                 AlwaysUseSubcells, SubcellToDgReconstructionMethod, UseHalo,
+                 OnlyDgBlocksAndGroups>;
 
   static constexpr Options::String help{
       "System-agnostic options for the DG-subcell method."};
 
   SubcellOptions() = default;
-  SubcellOptions(double initial_data_rdmp_delta0,
-                 double initial_data_rdmp_epsilon, double rdmp_delta0,
-                 double rdmp_epsilon, double initial_data_persson_exponent,
-                 double persson_exponent, bool always_use_subcells,
-                 fd::ReconstructionMethod recons_method, bool use_halo);
+  SubcellOptions(
+      double initial_data_rdmp_delta0, double initial_data_rdmp_epsilon,
+      double rdmp_delta0, double rdmp_epsilon,
+      double initial_data_persson_exponent, double persson_exponent,
+      bool always_use_subcells, fd::ReconstructionMethod recons_method,
+      bool use_halo,
+      std::optional<std::vector<std::string>> only_dg_block_and_group_names);
+
+  /// \brief Given an existing SubcellOptions that was created from block and
+  /// group names, create one that stores block IDs.
+  ///
+  /// The `DomainCreator` is used to convert block and group names into IDs
+  /// and also to check that all listed block names and groups are in the
+  /// domain.
```

Comment:
Maybe add a note that this is a workaround since we can't retrieve option tags from elsewhere in the input file?